ocaml/xenopsd/xc/service.ml

```diff
@@ -803,23 +781,15 @@ module Swtpm = struct
       absolute_path_outside chroot (Path.of_string ~relative:"swtpm-sock")
     )
 
-  let suspend _dbg ~xs ~domid ~vm_uuid ~vtpm_uuid:_ =
+  let suspend dbg ~xs ~domid ~vtpm_uuid =
     D.stop ~xs domid ;
-    Xenops_sandbox.Swtpm_guard.read ~domid ~vm_uuid state_path
+    Varstore_privileged_client.Client.vtpm_get_contents dbg vtpm_uuid
 
   let stop dbg ~xs ~domid ~vm_uuid ~vtpm_uuid =
     debug "About to stop vTPM (%s) for domain %d (%s)"
       (Uuidm.to_string vtpm_uuid)
       domid vm_uuid ;
-    let contents = suspend dbg ~xs ~domid ~vm_uuid ~vtpm_uuid in
-    let length = String.length contents in
-    if length > 0 then (
-      debug "Storing vTPM state of %d bytes" length ;
-      Varstore_privileged_client.Client.vtpm_set_contents dbg vtpm_uuid
-        (Base64.encode_string contents)
-    ) else
-      debug "vTPM state is empty: not storing" ;
-    (* needed to save contents before wiping the chroot *)
+    D.stop ~xs domid ;
```

Comment:
This wasn't stopped before (!?)

Reply:
Previously stop would call 'suspend' which would call D.stop.